### PR TITLE
include RFC 7694 (#119)

### PR DESCRIPTION
**Pull request by reschke**
This includes the extension, but leaves out:

- examples
- deployment considerations
- security considerations (but see https://github.com/httpwg/http-core/issues/6)

### CODE REVIEW

File: draft-ietf-httpbis-semantics-latest.xml [hunk 1]
```diff
@@ -5708,9 +5725,20 @@ Expect: 100-continue
...
-   target="content.codings"/>). An "identity" token is used as a synonym for
+   The "Accept-Encoding" header field can be used to indicate preferences
+   regarding the use of content-codings (<xref target="content.codings"/>).
+</t>
+<t>
+   When used by a user agent in a request, it applies to the content codings
+   acceptable in the response.
+</t>
+<t>
+   On the other hand, when used by a server in a response, it provides
```
Comment (mnot) [inline draft-ietf-httpbis-semantics-latest.xml]: nit: "On the other hand" is a bit wordy

File: draft-ietf-httpbis-semantics-latest.xml [hunk 2]
```diff
@@ -5759,6 +5787,37 @@ Expect: 100-continue
    is listed as acceptable, the origin server &SHOULD; send a response
    without any content-coding.
 </t>
+<t>
+   When the Accept-Encoding header field is present in a response, it indicates
+   what content codings the resource was willing to accept in the associated
+   request. The field value is evaluated the same way as in a request.
```
Comment (mnot) [inline draft-ietf-httpbis-semantics-latest.xml]: "associated request" doesn't seem appropriate here; we use that elsewhere to mean the request in a specific exchange, and so by definition this information can't ever be used (unless we want to get into the exact semantics of what a repeated / retried request is -- but that's another windmill).

I'd suggest something like "... what content codings the resource is currently willing to accept in requests similar to that associated with the response." 

And below: "Note that this information is a hint only; the server's willingness to accept an encoding can be influenced by many factors, including resource identity, time, request method, and other factors."

File: draft-ietf-httpbis-semantics-latest.xml [hunk 3]
```diff
@@ -5759,6 +5787,37 @@ Expect: 100-continue
...
+   include an Accept-Encoding header field in that response, allowing
+   clients to distinguish between issues related to content codings and media
+   types. In order to avoid confusion with issues related to media types,
+   servers that fail a request with a 415 status for reasons unrelated to
+   content codings <bcp14>MUST NOT</bcp14> include the Accept-Encoding header
+   field.
+</t>
+<t>
+   The most common use of Accept-Encoding is in responses with a
+   <x:ref>415 (Unsupported Media Type)</x:ref> status code, in response to
```
Comment (mnot) [inline draft-ietf-httpbis-semantics-latest.xml]: Maybe it's time for a new status code...

File: draft-ietf-httpbis-semantics-latest.xml (hunk 1 above)
Comment (reschke) [reply draft-ietf-httpbis-semantics-latest.xml]: just remove it?

File: draft-ietf-httpbis-semantics-latest.xml (hunk 2 above)
Comment (reschke) [reply draft-ietf-httpbis-semantics-latest.xml]: But that's exactly what it means here (the request leading to the response being discussed)...

I'm ok with the rephrased note, but it really doesn't seem to say anything different (the current text in the PR is from <https://greenbytes.de/tech/webdav/rfc7694.html#rfc.section.3.p.3>.

File: draft-ietf-httpbis-semantics-latest.xml (hunk 1 above)
Comment (mnot) [reply draft-ietf-httpbis-semantics-latest.xml]: I think so.

File: draft-ietf-httpbis-semantics-latest.xml (hunk 2 above)
Comment (mnot) [reply draft-ietf-httpbis-semantics-latest.xml]: I see. It might be good to explicitly state something like "Clients can use this information to optimistically encode future requests. However, because this information is specific to the associated request..."

Comment (reschke) [reply draft-ietf-httpbis-semantics-latest.xml]: Not sure here. Clients always could use content-codings in requests. All we add here is some signalling from server to client that gives more information. And the paragraph already clearly says what it applies to, and when it could change.

Can you propose some concrete text change?

**Comment by reschke** [comment]
Hm. My understanding is that content encodings are fully bidirectional (and have been all the time in theory), thus this would be exactly the right thing to do.

In your proposal, would we have two different sections for Accept-Encoding (request and response)?

**Comment by royfielding** [comment]
What I should have said is that defining a bunch of information about response header fields inside the main section on request header fields doesn't work. Moving it to a new section within the general discussion of content negotiation might work.

I have gone ahead and done that here. The section on Accept-Encoding still contains both request and response definitions, which seems fine.

File: draft-ietf-httpbis-semantics-latest.xml [hunk 4]
```diff
@@ -4236,6 +4240,57 @@ Content-Range: exampleunit 11.2-14.3/25
...
+    which content types are accepted in PATCH requests.
+  </t>
+</aside>
+</section>
+
+<section title="Quality Values" anchor="quality.values">
+  <x:anchor-alias value="weight"/>
+  <x:anchor-alias value="qvalue"/>
+<t>
+   The content negotiation fields
```
Comment (mnot) [inline draft-ietf-httpbis-semantics-latest.xml]: Either "The content negotiation fields defined by this specification" or "some content negotiation fields."

File: draft-ietf-httpbis-semantics-latest.xml [hunk 5]
```diff
@@ -4096,12 +4098,14 @@ Content-Range: exampleunit 11.2-14.3/25
...
    be made visible within the protocol:
-   "proactive", where the server selects the representation based
-   upon the user agent's stated preferences, and "reactive" negotiation,
+   "proactive" negotiation, where the server selects the representation based
+   upon the user agent's stated preferences, "reactive" negotiation,
    where the server provides a list of representations for the user agent to
-   choose from. Other patterns of content negotiation include
+   choose from, and "request payload" negotiation, where the user agent
+   selects the representation for a future request based upon the server's
+   stated preferences. Other patterns of content negotiation include
```
Comment (mnot) [inline draft-ietf-httpbis-semantics-latest.xml]: "stated preferences in past responses." ?

**Comment by mnot** [comment]
Looks better, left two small-ish comments.

File: draft-ietf-httpbis-semantics-latest.xml (hunk 4 above)
Comment (reschke) [reply draft-ietf-httpbis-semantics-latest.xml]: done

File: draft-ietf-httpbis-semantics-latest.xml (hunk 5 above)
Comment (reschke) [reply draft-ietf-httpbis-semantics-latest.xml]: done
